### PR TITLE
Migrate From Node-Sass to Dart-Sass

### PR DESCRIPTION
**Pull request by reldredge71**
Node-Sass is becoming deprecated; it is recommended to migrate to Dart-Sass.

These changes are required along with this migration:
https://sass-lang.com/documentation/breaking-changes/extend-compound
https://sass-lang.com/documentation/breaking-changes/slash-div
https://github.com/foundation/foundation-rails/issues/296

### CODE REVIEW

**Comment by reldredge71** [comment]
@wrandall22 @OzzieOrca @TheNoodleMoose I'm still getting deprecation warnings for node_modules.  the `quietDeps` option should disable those, but it has not worked yet.  It's possible that there is some issue with using`quietDeps` along with `MiniCssExtractPlugin`: https://github.com/sass/sass/issues/3065#issuecomment-910946898

File: src/assets/cru-scss/_subnav.scss
Comment (OzzieOrca) [inline src/assets/cru-scss/_subnav.scss:1]: You using math here?

File: src/assets/cru-scss/_supernav.scss
Comment (OzzieOrca) [inline src/assets/cru-scss/_supernav.scss:1]: Same here, I don't see math being used. (maybe an unchanged line is using it, I only scrolled through the whole first file).

File: src/assets/crubrand/generic/_tools.scss
Comment (OzzieOrca) [inline src/assets/crubrand/generic/_tools.scss:27]: I don't think order of operations affects this but I probably would have had the ` * 1rem` after the `math.div` function call instead of as the second parameter.

File: src/assets/crubrand/generic/_tools.scss
Comment (OzzieOrca) [inline src/assets/crubrand/generic/_tools.scss:80]: Same here.

File: src/assets/crubrand/generic/mixins/_media-query.scss
Comment (OzzieOrca) [inline src/assets/crubrand/generic/mixins/_media-query.scss:22]: What does `list.slash` do? This looks like a division...

File: src/assets/crubrand/generic/mixins/_sizing.scss
Comment (OzzieOrca) [inline src/assets/crubrand/generic/mixins/_sizing.scss:34]: I think you have the order of operations here wrong. https://sass-lang.com/documentation/operators#order-of-operations

**Review by OzzieOrca — CHANGES_REQUESTED** [review]
Thanks for updating all those. That was a lot of CSS to go through :)

Idk what's going on with `quietDeps`... The config looks right... https://sass-lang.com/documentation/js-api/interfaces/Options#quietDeps I wonder if somehow the directory config is messed up for this project and it's not identifying `node_modules` right. I could do some digging but I'm not sure what to suggest right now...

File: src/assets/cru-scss/_subnav.scss
Comment (reldredge71) [reply src/assets/cru-scss/_subnav.scss:1]: Hmm I must've had a reason for adding it, but I thought I only added it to files that I was removing the `/` from.  Like you said, it was a lot of CSS to go through!  I'll remove it.

File: src/assets/crubrand/generic/mixins/_media-query.scss
Comment (reldredge71) [reply src/assets/crubrand/generic/mixins/_media-query.scss:22]: Yes, it should behave the same way as a division.  I got an error saying that `math.div` should only be used for number variables, and I guess it didn't know how to process the results of `strip-units`.  then again, `strip-units` is used in other places and it doesn't seem to cause an issue, I'll have to look into that.

I'm actually noticing another order-of-operations issue here -- I assumed that `+ em` was part of the right side of the `/` operator, but now I realize that it should be it's own thing.

File: src/assets/crubrand/generic/mixins/_sizing.scss
Comment (reldredge71) [reply src/assets/crubrand/generic/mixins/_sizing.scss:34]: yeah, I'm realizing now that I kept thinking of the additions as part of the right-side of the `/` operator.  I know order of operations, but I guess I was imagining parentheses around `$sizeValue + rem`.

**Review by OzzieOrca — DISMISSED** [review]

File: src/assets/crubrand/generic/mixins/_media-query.scss
Comment (wrandall22) [inline src/assets/crubrand/generic/mixins/_media-query.scss:1]: This appears to be unused now.

**Comment by wrandall22** [comment]
Make sure to merge to staging and test there before merging to master.

**Comment by reldredge71** [comment]
I wish I could quiet the warnings from node_modules, but at least they don't hold it back from building properly.

I'm noticing one specific layout issue after this migration.  Some items are spilling over the right edge of their containers.  I inspected these elements compared to give-stage2.cru.org, and the only thing that stands out to me is that these elements are missing the CSS property `box-sizing: border-box` post-migration.  I would like to figure out why that is before merging to staging.

Screen Shot 2022-01-28 at 3 35 22 PM

Screen Shot 2022-01-28 at 3 35 00 PM

File: src/assets/crubrand/generic/bourbon/addons/_prefixer.scss
Comment (reldredge71) [inline src/assets/crubrand/generic/bourbon/addons/_prefixer.scss:10]: I'm trying to find the right way to setup these variables.  I think they're supposed to be loaded from src/assets/crubrand/generic/bourbon/settings/_prefixer.scss.  However, I was noticing strange behavior in the layout of some elements, which I traced back to the prefixer mixin in this file, and explicitly declaring these variable defaults in this file fixed the issue.

On a related note, I'm getting deprecation warnings for this file as well, in the disable-prefix-for-all mixin:

Screen Shot 2022-02-04 at 9 51 30 AM

It's complaining about this same list of variables, and about trying to assign them globally.  I can look into this deeper, but if anyone has any insight on it I would appreciate it.

Comment (reldredge71) [reply src/assets/crubrand/generic/bourbon/addons/_prefixer.scss:10]: BTW these changes that I made to `_prefixer.scss` fixed the weird `box-sizing` issue I mention above.

Comment (wrandall22) [reply src/assets/crubrand/generic/bourbon/addons/_prefixer.scss:10]: Are there adverse effects to removing the `@mixin disable-prefix-for-all() {` block? I'm guessing it was put there for a reason, but I was not part of this codebase when it was being written.

Comment (reldredge71) [reply src/assets/crubrand/generic/bourbon/addons/_prefixer.scss:10]: It looks like it gets referenced here: https://github.com/CruGlobal/give-web/blob/master/src/assets/crubrand/generic/bourbon/css3/_keyframes.scss.  That file also seems to try to reference the same variables declared in https://github.com/CruGlobal/give-web/blob/master/src/assets/crubrand/generic/bourbon/settings/_prefixer.scss.

Comment (reldredge71) [reply src/assets/crubrand/generic/bourbon/addons/_prefixer.scss:10]: If that's the case, I could try copying the default values there as well, but it would probably be better if they were defined in one place that both files can access.

Comment (wrandall22) [reply src/assets/crubrand/generic/bourbon/addons/_prefixer.scss:10]: https://stackoverflow.com/questions/60422903/cannot-make-variables-globally-accessible-in-sass

**Comment by wrandall22** [comment]
Do the updates still have the problems resolved?

**Comment by reldredge71** [comment]
Yes, what you suggested worked great!  Sorry, I should have made that clear.

**Comment by wrandall22** [comment]
You can merge this to staging if you want

**Comment by reldredge71** [comment]
This is looking good on staging to me

**Review by wrandall22 — APPROVED** [review]

**Pull request merged**
